lib/host/wasi/inode-linux.cpp

```diff
@@ -273,14 +273,14 @@ INode::fdFilestatGet(__wasi_filestat_t &Filestat) const noexcept {
     return WasiUnexpect(Res);
   }
 
-  Filestat.dev = Stat->st_dev;
-  Filestat.ino = Stat->st_ino;
+  Filestat.dev = isSpecialFd(Fd) ? 0 : Stat->st_dev;
```

Comment:
We should have a comment section explaining why the special fds need this handling way.